oci/layout/oci_transport.go

```diff
@@ -194,10 +203,14 @@ func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
 		}
 	} else {
 		// if image specified, look through all manifests for a match
-		for _, md := range index.Manifests {
+		for i, md := range index.Manifests {
 			if md.MediaType != imgspecv1.MediaTypeImageManifest && md.MediaType != imgspecv1.MediaTypeImageIndex {
 				continue
 			}
+			if ref.sourceIndex == i {
```

Comment:
Why not special-case this and index `index.Manifests[ref.sourceIndex]` directly? That would also allow returning a detailed error (“index too large, only %d entries available”) instead of just a “not found”.